### PR TITLE
Prepare for release 0.59.0

### PR DESCRIPTION
**Pull request by atoulme**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,9 +4,14 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/).
 
 ## Unreleased
 
-### Added
+## [0.59.0] - 2022-09-17
+
+### Changed
 
-- A way to provide a custom image for init container patching host log directories (#534, #535)
```
Comment (jvoravong) [inline CHANGELOG.md:9]: Should we delete this line?

**Review by jvoravong — APPROVED** [review]
Once comment. LGTM.

File: helm-charts/splunk-otel-collector/Chart.yaml [hunk 2]
```diff
@@ -1,7 +1,7 @@
 apiVersion: v2
 name: splunk-otel-collector
-version: 0.58.0
-appVersion: 0.58.0
+version: 0.59.0
+appVersion: 0.59.0
```
Comment (rmfitzpatrick) [inline helm-charts/splunk-otel-collector/Chart.yaml:4]: should this be 0.59.1?

File: CHANGELOG.md (hunk 1 above)
Comment (atoulme) [reply CHANGELOG.md:9]: I mangled this line, sorry. Fixed.

File: helm-charts/splunk-otel-collector/values.yaml [hunk 3]
```diff
@@ -829,7 +829,7 @@ image:
     # The registry and name of the opentelemetry collector image to pull
     repository: quay.io/signalfx/splunk-otel-collector
     # The tag of the Splunk OTel Collector image, default value is the chart appVersion
-    tag: ""
+    tag: "0.59.1"
```
Comment (jvoravong) [inline helm-charts/splunk-otel-collector/values.yaml:832]: This shouldn't bed needed after addressing [rmfitzpatrick](https://github.com/rmfitzpatrick)'s comment.

File: helm-charts/splunk-otel-collector/Chart.yaml (hunk 2 above)
Comment (atoulme) [reply helm-charts/splunk-otel-collector/Chart.yaml]: Missed that one. Fixing.

File: helm-charts/splunk-otel-collector/values.yaml (hunk 3 above)
Comment (atoulme) [reply helm-charts/splunk-otel-collector/values.yaml:832]: I'll revert right now.

**Pull request merged**